### ISSUE TITLE
Should Client::send() reset the responses?

### ISSUE THREAD

**Issue by shtse8** [ptcong/easyrequest#7]
Problem: A request with multiple send() results in multiple responses.

Situation: I request a site through proxy. due to the instability of the proxy, the proxy got unexpected response. I will resend the request using the same request.

Suggestion: send() should reset the responses to represent the responses of each send.

**Comment by shtse8** [comment]
#8 will also happen if multiple send() are used in the same request. Because when send() is called. Cookie will be added to the request, on the second time prepareRequest(), cookie is added to the cookieJar again without trimming the cookie name.

**Comment by ptcong** [comment]
Why don't just set requests property to an empty array when resend request ?
i think you should make a clone of current request object and resend it it current is failed in your case, dont need to modify the script for this case.

**Comment by shtse8** [comment]
Makes sense. but developers may not expect the object can only be used once and think that they are making a request using this class and try to "send" the request. they also will try to call send() multiple times without getting any exception and warnings but errors response and wrong cookie behavior.
So I suggest this class should support multiple send() and let the object can be re-used.
